text/chapter7.md

```diff
@@ -109,6 +109,8 @@ class Functor f <= Apply f where
 
 The `Apply` type class is a subclass of `Functor`, and defines an additional function `apply`. As `<$>` was defined as an alias for `map`, the `Prelude` module defines `<*>` as an alias for `apply`. As we'll see, these two operators are often used together.
 
+Note that this [`apply`](https://pursuit.purescript.org/packages/purescript-prelude/docs/Control.Apply#v:apply) is different than the [`apply`](https://pursuit.purescript.org/packages/purescript-prelude/docs/Data.Function#v:apply) from `Data.Function` (infixed as `$`). Luckily, infix notation is almost always used for the latter, so you don't need to worry about name collisions. And yes, it may have been nicer to swap the `<*>` and `<$>` operators so that each `apply` involved the dollar-sign; however, we wanted to keep our notation consistent with Haskell.
```

Comment:
The dollar-sign in `<$>` (`map`) already resembles `$` &mdash; applying a function _in_ a Functor context. Consider the typical usage `f <$> m1 <*> m2` where `f :: a -> b -> c` and `m1 :: m a` and `m2 :: m b`. `<*>` is applying a function _already in_ context. We may have wished it be the slightly longer `<$$>` to also resemble `$` but choosing a different single special character was probably better for terseness. I would argue that `map` more closely resembles function application and wins `<$>`.

There's probably more mathematical historic reasons on these choices. Like why did they pick `$` for function application in the first place.

Anyhow, I don't think we should suggest that one should wish the operator symbols match their _named_ functions. But explain why it doesn't. And didn't in Haskell.

Reply:
> I would argue that `map` more closely resembles function application and wins `<$>`.

That's a good point. Now it seems like the choice of names for these functions isn't ideal, but that's understandable, since naming stuff is hard. Table for future reference:
```purs
apply  $  ::   (a -> b) ->   a ->   b
map   <$> ::   (a -> b) -> f a -> f b
apply <*> :: f (a -> b) -> f a -> f b
```

> But explain why it doesn't. And didn't in Haskell.

I'm planning to just drop this additional note from the PR. I'd be interested in reading how Haskell selected these operators, but haven't found any material on that yet.